### DISCUSSION TITLE
Announcing a toolkit for embedding v3 Studio inside Next.js apps

### DISCUSSION THREAD

**Discussion by stipsan** [sanity-io/sanity#3479] [Announcements]
We've just released [`next-sanity/studio`](https://github.com/sanity-io/next-sanity#next-sanitystudio-dev-preview). A "batteries included" experience for when you're using your Studio in a Next.js environment.

## Why is this toolkit needed?

Using Studio v3 inside of Next.js isn't as simple  as just creating a `pages/studio/[[...index]].tsx` and then it's good to go:
```tsx
import {Studio} from 'sanity'
import config from '../../sanity.config'

export default function StudioPage() {
  return <Studio config={config} />
}
```

There are two important things you must setup for it to work well:
1. `html,body,#__next` parent wrappers need to have `height: 100%` for the Studio to expand to fill the viewport. Compare how it looks [without](https://user-images.githubusercontent.com/81981/183253434-965daf11-65fc-49b9-96af-a2e2a005383e.png) and [with](https://user-images.githubusercontent.com/81981/183253436-d8fa0a80-96f1-4f54-8d63-55af13ca3d02.png) that CSS.
2. `pages/_document.tsx` needs to be [setup to handle styled-components SSR](https://github.com/vercel/next.js/blob/canary/examples/with-styled-components/pages/_document.tsx). If you turn off JS and load the Studio it becomes very noticeable why. [Without](https://user-images.githubusercontent.com/81981/183253433-f967f755-9c1d-4acf-bed0-f17d9e6f8b07.png) SSR setup, and [with](https://user-images.githubusercontent.com/81981/183253437-1396d22f-a40f-4d2d-aab6-88ad600a700b.png) `_document.tsx`.

The new toolkit provides two components for you that solve both these problems (and more!) for you:
- `NextStudio`, a drop-in replacement for `Studio` in your `pages/studio/[[...index]].tsx`
   ```tsx
   import {NextStudio} from 'next-sanity/studio'
   import config from '../../sanity.config'

   export default function StudioPage() {
     return <NextStudio config={config} />
   }
   ```
- `ServerStyleSheetDocument`, a wrapper for `next/document` that implements the SSR, here's how difficult it is to use:
   ```tsx
   // pages/_document.tsx
   import {ServerStyleSheetDocument} from 'next-sanity/studio'

   export default class Document extends ServerStyleSheetDocument {}
   ```

😮‍💨

We got a few examples using it already, with more on the way:
- [Simple, as in using only defaults](https://github.com/sanity-io/themer/tree/main/examples/next-static)
- [Advanced, it sets a `modulepreload` link to `<head>`](https://github.com/sanity-io/themer/tree/main/examples/next-dynamic)
- [Darksouls mode, everything is so custom that it only uses two of the components that `NextStudio` composes](https://github.com/sanity-io/themer/blob/e370240dabbffa65e0c88ea9225d2b2542146d51/components/Head.tsx)
- And finally, `next-sanity/studio` [dogfoods itself](https://github.com/sanity-io/next-sanity/blob/2c97f35d2228a0f32d5d6262de023087b3d1ad2c/pages/%5B%5B...studio%5D%5D.tsx). [Checkout the production deploy](https://next.sanity.build/).

**Comment by wer-mathurin** [comment]
@stipsan
Any plan to hand the same kind of integration for Flutter(web)
